source/configuration/modules/imfile.rst

```diff
@@ -1,570 +1,408 @@
 
-   This is used to turn on or off the addition of metadata to the
-   message object.
+================  ======================
+pollinginterval:  "seconds"
+Default:          "10"
+Legacy:           $InputFilePollInterval
+Available since:  8.1.5
+================  ======================
 
-.. function:: stateFile [name-of-state-file]
+This setting specifies how often files are checked for changes when using **mode="polling"**. Basically, this mean files are read and rsyslog sleeps for X seconds before being read again.
 
-   **Default: unset**
+Although it's possible, its stongly recommend not to set it to 0 seconds, because in such scenario rsyslog will take far too many resources.
 
-   **This paramater is deprecated.** It still is accepted, but should
-   no longer be used for newly created configurations.
+Input parameters
+****************
 
-   This is the name of this file's state file. This parameter should
-   usually **not** be used. Check the section on "State Files" above
-   for more details.
-
-.. index::
-   single: imfile; reopenOnTruncate
-.. function:: reopenOnTruncate [on/off] (requires v8.16.0+)
-
-   **Default: off**
-
-   This is an **experimental** feature that tells rsyslog to reopen input file
-   when it was truncated (inode unchanged but file size on disk is less than
-   current offset in memory).
-
-.. index::
-   single: imfile; trimLineOverBytes
-.. function:: trimLineOverBytes [number] (requires v8.17.0+)
+============  ===============
+**REQUIRED**
+file:         "/path/to/file"
+Default:      none
+Legacy:       $InputFileName
+============  ===============
 
-   **Default: 0**
+File(s) being processed. This must be an absolute file path (no macros or templates). 
 
-   This is used to tell rsyslog to truncate the line which length is greater
-   than specified bytes. If it is positive number, rsyslog truncate the line
-   at specified bytes. Default value of 'trimLineOverBytes' is 0, means never
-   truncate line.
+This option currently supports *file wilcards* in **inotify mode only** (it hasn't been implemented for polling) for monitoring multiple files. Consider wildcards are only valid at directory level, ie:
 
-   This option can be used when ``readMode`` is 0 or 2.
+* /path/to/file.log is valid and will monitor file named file.log
+* /path/to/file\*.log is valid and will monitor all files named like *file\**.log
+* /path/to/\*name.log is valid and will monitor all files named like *\*name*.log
+* /path/to\*/file.log is invalid
+* /path/to/foo being foo a directory is also invalid
 
-.. index::
-   single: imfile; freshStartTail
-.. function:: freshStartTail [on/off] (requires v8.18.0+)
+============  =============
+**REQUIRED**
+tag:          "value"
+Default:      none
+Legacy:       $InputFileTag
+============  =============
 
-   **Default: off**
+**TODO: preperly link Syslog Specification https://tools.ietf.org/html/rfc5424#section-6.2.5**
```

Comment:
just put this link in the doc. People almost never need to reference it, so I think we are good just putting a link here, even for people reading this without a network connection.